[PATCH] From afff474c79e0c177e090d1b45d68c9f816a52e3e Mon Sep 17

From: Pavel Shirshov <pavelsh@microsoft.com>
Date: Mon, 16 Nov 2020 18:33:46 -0800
 00:00:00 2001 Subject: [PATCH 4/8] Allow BGP attr NEXT_HOP to be 0.0.0.0 due
 to  alleviate
---
 bgpd/bgp_route.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/bgpd/bgp_route.c b/bgpd/bgp_route.c
index aa6c701ca..a8427bc3b 100644
--- a/bgpd/bgp_route.c
+++ b/bgpd/bgp_route.c
@@ -4786,8 +4786,7 @@ bool bgp_update_martian_nexthop(struct bgp *bgp, afi_t afi, safi_t safi,
 	 * E.g.: IPv6 prefix can be with nexthop: 0.0.0.0, and mp_nexthop: fc00::1.
 	 */
 	if (CHECK_FLAG(attr->flag, ATTR_FLAG_BIT(BGP_ATTR_NEXT_HOP)))
-		nh_invalid = (attr->nexthop.s_addr == INADDR_ANY ||
-			      !ipv4_unicast_valid(&attr->nexthop) ||
+		nh_invalid = (!ipv4_unicast_valid(&attr->nexthop) ||
 			      bgp_nexthop_self(bgp, afi, type, stype, attr, dest));
 
 	/* If MP_NEXTHOP is present, validate it. */
